examples/extra_element_integer/random_cluster.cpp

```diff
@@ -0,0 +1,128 @@
+#include "libmesh/libmesh.h"
+#include "libmesh/mesh.h"
+#include "libmesh/exodusII_io.h"
+#include "libmesh/elem.h"
+#include "libmesh/equation_systems.h"
+#include "libmesh/linear_implicit_system.h"
+#include "libmesh/dof_map.h"
+#include "libmesh/numeric_vector.h"
+#include "libmesh/system.h"
+#include "libmesh/mesh_generation.h"
+
+#include <vector>
+#include <cstdlib>
+#include <ctime>
+
+using namespace libMesh;
+
+void CreateMesh(libMesh::Mesh & mesh, int nx, int ny){
+
+    libMesh::MeshTools::Generation::build_square(mesh,
+                                                 nx,
+                                                 ny,
+                                                 0.0, 10.0,
+                                                 0.0, 10.0,
+                                                 libMesh::QUAD4);
+}
+
+void CreateMesh(libMesh::Mesh & mesh, int nx, int ny, int nz ){
+    libMesh::MeshTools::Generation::build_cube(mesh,
+                                               nx,
+                                               ny,
+                                               nz,
+                                               0.0, 10.0,
+                                               0.0, 10.0,
+                                               0.0, 10.0);
+}
+
+void PopulateRandomIntegers(libMesh::Mesh & mesh, LinearImplicitSystem& system, const unsigned int &index ) {
+
+    DofMap& dof_map = system.get_dof_map();
+    for (const auto& elem : mesh.element_ptr_range()) {
+
+        std::vector<dof_id_type> dof_indices;
+        dof_map.dof_indices(elem, dof_indices);
+        int random_int = rand() %5;
+
+        system.solution->set(dof_indices[1], random_int);
+        elem->set_extra_integer(index, 5-random_int);
+
+    }
+}
+
+void AddVaribalesToSystem(LinearImplicitSystem& system, libMesh::EquationSystems & equation_systems){
+
+    system.add_variable("cluster_id", CONSTANT, MONOMIAL);
+    system.add_variable("random_field", CONSTANT, MONOMIAL);
+
+    equation_systems.init();
+}
+
+DofMap& CreateDegreesofFreedomMap(LinearImplicitSystem& system){
+    return  system.get_dof_map();
+}
+
+void FindCluster(libMesh::Mesh & mesh, libMesh::DofMap & dof_map, LinearImplicitSystem& system, const unsigned int index ) {
```

Comment:
I think this method is assuming the wrong things about the existing data.  It should only assume that there is field data in a variable of an equation system and nothing else.  It should then use some algorithm - it can be trivial for the moment - that accesses that data to establish a value for the cluster ID.